### PR TITLE
PR template: shorten time to complete & add rust-vmm checkbox

### PR DESCRIPTION
**Pull request by aghecenco**
## Reason for This PR

The PR checklist is useful, but at times it's cumbersome to use, particularly because of the manual step of editing the `A or B` items (then fixing alignment... :woman_facepalming:). This PR eliminates the manual editing step, shortens the checklist items' texts, and adds a new item, not mandatory to check, meant to maintain awareness of functionality that can be shared in the broader context of `rust-vmm`.

#1231 will shorten the checklist even more (the data shows that checklist length is not directly proportional to productivity) :rocket: 

## Description of Changes

* collapsed `A or B` items in non-editable, single-phrased items
* shortened long texts
* **gave up on the 80 character limit because GitHub inserts a newline and the author would need to manually edit it out**
* added `rust-vmm` integration checkbox

- [ ] This functionality can be added in [`rust-vmm`](https://github.com/rust-vmm).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria. Where there are two options, keep one.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The reason for this PR is clearly provided (issue no. or explanation).
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] Any newly added `unsafe` code is properly documented.
- [x] Any API changes are reflected in `firecracker/swagger.yaml`.
- [x] Any user-facing changes are mentioned in `CHANGELOG.md`.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -13,21 +13,14 @@ the terms of the Apache 2.0 license.
 
 ## PR Checklist
 
-`[Author TODO: Meet these criteria. Where there are two options, keep one.]`
+`[Author TODO: Meet these criteria.]`
 `[Reviewer TODO: Verify that these criteria are met. Request changes if not]`
 
+- [ ] This functionality can be added in [`rust-vmm`](https://github.com/rust-vmm).
```
Comment (acatangiu) [inline .github/pull_request_template.md]: This looks out of place here, it's not really a criteria the PR should meet.

I'd add it in its own category.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -13,21 +13,14 @@ the terms of the Apache 2.0 license.
...
-- [ ] Either no API changes are included in this PR, or, the API changes are
-      reflected in `firecracker/swagger.yaml`.
-- [ ] Either the changes in this PR have no user impact, or, the changes in
-      this PR have user impact and have been added to the `CHANGELOG.md` file.
-- [ ] Either no new `unsafe` code has been added, or, the newly added `unsafe`
-      code is unavoidable and properly documented.
+- [ ] Any required documentation changes (code and docs) are included in this PR.
+- [ ] Any newly added `unsafe` code is properly documented.
+- [ ] Any API changes are reflected in `firecracker/swagger.yaml`.
+- [ ] The `CHANGELOG.md` file is up to date.
```
Comment (acatangiu) [inline .github/pull_request_template.md]: Maybe leave a few words to indicate only user-facing changes should go in the CHANGELOG.

Comment (raduiliescu) [reply .github/pull_request_template.md]: Like:
Any user-facing changes contained in this PR are reflected in CHANGELOG.md

Also I would put Any API changes *contained in this PR* are reflected in `firecracker/swagger.yaml`.

File: .github/pull_request_template.md (hunk 1 above)
Comment (aghecenco) [reply .github/pull_request_template.md]: Moved it to the `## Description of Changes` category - maybe while the contributor is writing the description, they can be prompted this way to consider sharing on the broader context.

File: .github/pull_request_template.md (hunk 2 above)
Comment (aghecenco) [reply .github/pull_request_template.md]: Reworded.

**Pull request merged**